diff --git a/libraries/extractor/src/main/java/androidx/media3/extractor/ts/H265Reader.java b/libraries/extractor/src/main/java/androidx/media3/extractor/ts/H265Reader.java
--- a/libraries/extractor/src/main/java/androidx/media3/extractor/ts/H265Reader.java
+++ b/libraries/extractor/src/main/java/androidx/media3/extractor/ts/H265Reader.java
@@ -256,7 +256,9 @@ public final class H265Reader implements ElementaryStreamReader {
 
     // Skip the 3-byte NAL unit start code synthesised by the NalUnitTargetBuffer constructor.
+    NalUnitUtil.H265VpsData vpsData =
+        NalUnitUtil.parseH265VpsNalUnit(vps.nalData, /* nalOffset= */ 3, vps.nalLength);
     NalUnitUtil.H265SpsData spsData =
         NalUnitUtil.parseH265SpsNalUnit(
-            sps.nalData, /* nalOffset= */ 3, sps.nalLength, /* vpsData= */ null);
+            sps.nalData, /* nalOffset= */ 3, sps.nalLength, vpsData);
 
     @Nullable String codecs = null;
